src/btree/bt_page.c

```diff
@@ -65,6 +65,9 @@ __wt_page_alloc(WT_SESSION_IMPL *session,
 	page->type = type;
 	page->read_gen = WT_READGEN_NOTSET;
 
+	/* Initialize the page lock. */
+	__wt_rwlock_init(session, &page->page_lock);
+
```

Comment:
This is a real bug regardless of other changes: we were relying on `__wt_calloc` to initialize a `WT_RWLOCK`.